store/localcachelayer/channel_layer.go

```diff
@@ -73,6 +87,56 @@ func (s LocalCacheChannelStore) InvalidateMemberCount(channelId string) {
 	}
 }
 
+// removeByPrefixFromChannelMembersForUserCache removes all keys from
+// channelMembersForUserCache with the prefix of the given userId.
+func (s LocalCacheChannelStore) removeByPrefixFromChannelMembersForUserCache(userId string) {
```

Comment:
I would remove this logic into the new cache interface, this would allow to implement this logic in a better way depending on the backend. For example, we could use in redis something like `DELETE userId_*` or something like that, with this implementation you will end up generating 1 roundtrip with redis per key.

Reply:
Sounds great. :+1: